discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -451,102 +389,79 @@ impl SearchScope for SqliteStorage {
         search: &Search,
         documents: &[NewDocument],
     ) -> Result<(), Error> {
-        let mut tx = self.begin_tx().await?;
-        let mut query_builder = QueryBuilder::new("INSERT INTO ");
+        let mut tx = self.pool.begin().await?;
 
-        query_builder
-            .push(
-                "Search (rowid, searchBy, searchTerm, pageSize, pageNumber) VALUES (1, ?, ?, ?, ?)",
-            )
-            .build()
-            .persistent(false)
+        sqlx::query(
+            "INSERT INTO Search (rowid, searchBy, searchTerm, pageSize, pageNumber) VALUES (1, ?, ?, ?, ?)"
+        )
             .bind(search.search_by as u8)
             .bind(&search.search_term)
             .bind(search.paging.size)
             .bind(search.paging.next_page)
             .execute(&mut tx)
-            .await
-            .map_err(|err| Error::Database(err.into()))?;
+            .await?;
 
         if documents.is_empty() {
-            return Self::commit_tx(tx).await;
+            tx.commit().await?;
+            return Ok(());
```

Comment:
:art: this could be handled as a single statement
```suggestion
            return tx.commit().await.map_err(Into::into);
```